doc/index.rst

```diff
@@ -0,0 +1,189 @@
+currently `commons`, `core`, `brainsimulation`, `electrophysiology`, `software`, `minds` and `uniminds`.
+For example::
+
+   from fairgraph.commons import BrainRegion
+   from fairgraph.electrophysiology import PatchedCell
+
+Using these classes, it is possible to list all metadata matching a particular criterion, e.g.::
+
+   cells_in_ca1 = PatchedCell.list(client, brain_region=BrainRegion("hippocampus CA1"))
+
+If you know the unique identifier of an object, you can retrieve it directly::
+
+   cell_of_interest = PatchedCell.from_uuid("153ec151-b1ae-417b-96b5-4ce9950a3c56", client)
+
+Links between metadata in the Knowledge Graph are not followed automatically,
+to avoid unnecessary network traffic, but can be followed with the `resolve()` method::
+
+   example_cell = cells_in_ca1[3]
+   experiment = example_cell.experiments.resolve(client)
+   trace = experiment.traces.resolve(client)
+
+
+The associated metadata is accessible as attributes of the Python objects, e.g.::
+
+   print(example_cell.cell_type)
+   print(example_cell.reversal_potential_cl)
+   print(trace.time_step)
+   print(trace.data_unit)
+
+
+You can also access any associated data::
+
+   import requests
+   import numpy as np
+   from io import BytesIO
+
+   download_url = trace.data_location['downloadURL']
+   data = np.genfromtxt(BytesIO(requests.get(download_url).content))
+
+
+Advanced queries
+----------------
+
+While certain filters and queries are built in (such as the filter by brain region, above),
+more complex queries are possible using the Nexus query API.
+
+::
+
+   from fairgraph.base import KGQuery
+   from fairgraph.minds import Dataset
+
+   query = {
+      "path": "minds:specimen_group / minds:subjects / minds:samples / minds:methods / schema:name",
+      "op": "in",
+      "value": ["Electrophysiology recording",
+               "Voltage clamp recording",
+               "Single electrode recording",
+               "functional magnetic resonance imaging"]
+   }
+   context = {
+               "schema": "http://schema.org/",
+               "minds": "https://schema.hbp.eu/minds/"
+   }
+
+   activity_datasets = KGQuery(Dataset, query, context).resolve(client)
+   for dataset in activity_datasets:
+      print("* " + dataset.name)
+
+
+Storing and editing metadata
+----------------------------
+
+For those users who have the necessary permissions to store and edit metadata in the Knowledge Graph,
+**fairgraph** objects can be created or edited in Python, and then saved back to the Knowledge Graph, e.g.::
+
+   from fairgraph.core import Person, Organization
+   from fairgraph.commons import Address
+
+   mgm = Organization("Metro-Goldwyn-Mayer")
+   mgm.save(client)
+   author = Person("Laurel", "Stan", "laurel@example.com", affiliation=mgm)
+   author.save(client)
+
+::
+
+   mgm.address = Address(locality='Hollywood', country='United States')
+   mgm.save(client)
```

Comment:
I think this code is broken. I get:
```
Traceback (most recent call last):
  File "test.py", line 11, in <module>
    mgm.save(client)
  File "/home/yzerlaut/anaconda3/lib/python3.7/site-packages/fairgraph/base.py", line 474, in save
    if self.id or self.exists(client):
  File "/home/yzerlaut/anaconda3/lib/python3.7/site-packages/fairgraph/base.py", line 396, in exists
    response = client.query_nexus(self.__class__.path, query_filter, context)
  File "/home/yzerlaut/anaconda3/lib/python3.7/site-packages/fairgraph/base.py", line 100, in path
    raise ValueError("namespace not set")
ValueError: namespace not set
```

Reply:
I made an update of the documentation in a previous pull request to solve (/hide) this:
https://github.com/HumanBrainProject/fairgraph/pull/18/commits/2e3eb801e2fbbca3efdf1e4c6b59c8f661a8ab27